server/tests/src/test/java/org/infinispan/server/functional/ProtocolManagementIT.java

```diff
@@ -0,0 +1,92 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.infinispan.client.rest.IpFilterRule;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.commons.test.Exceptions;
+import org.infinispan.server.network.NetworkAddress;
+import org.infinispan.server.test.core.ServerRunMode;
+import org.infinispan.server.test.junit4.InfinispanServerRule;
+import org.infinispan.server.test.junit4.InfinispanServerRuleBuilder;
+import org.infinispan.server.test.junit4.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 12.0
+ **/
+public class ProtocolManagementIT {
+
+   @ClassRule
+   public static InfinispanServerRule SERVERS =
+         InfinispanServerRuleBuilder.config("configuration/MultiEndpointClusteredServerTest.xml")
+               .runMode(ServerRunMode.EMBEDDED)
+               .numServers(2)
+               .property("infinispan.bind.address", "0.0.0.0")
+               .build();
+
+   @Rule
+   public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+   @Test
+   public void testIpFilter() throws IOException {
+      NetworkAddress loopback = NetworkAddress.loopback("loopback");
+      RestClientConfigurationBuilder loopbackBuilder = new RestClientConfigurationBuilder();
+      loopbackBuilder.addServer().host(loopback.getAddress().getHostAddress()).port(11222);
+      RestClient loopbackClient = SERVER_TEST.rest().withClientConfiguration(loopbackBuilder).get();
+      assertEquals(200, sync(loopbackClient.server().connectorNames()).getStatus());
+
+      NetworkAddress siteLocal = NetworkAddress.siteLocal("sitelocal");
+      RestClientConfigurationBuilder siteLocalBuilder0 = new RestClientConfigurationBuilder();
+      siteLocalBuilder0.addServer().host(siteLocal.getAddress().getHostAddress()).port(11222);
+      RestClient siteLocalClient0 = SERVER_TEST.rest().withClientConfiguration(siteLocalBuilder0).get();
+      assertEquals(200, sync(siteLocalClient0.server().connectorNames()).getStatus());
+
+      RestClientConfigurationBuilder siteLocalBuilder1 = new RestClientConfigurationBuilder();
+      siteLocalBuilder1.addServer().host(siteLocal.getAddress().getHostAddress()).port(11322);
+      RestClient siteLocalClient1 = SERVER_TEST.rest().withClientConfiguration(siteLocalBuilder1).get();
+      RestResponse response = sync(siteLocalClient1.server().connectorNames());
+      assertEquals(200, response.getStatus());
+
+      List<IpFilterRule> rules = new ArrayList<>();
+      rules.add(new IpFilterRule(IpFilterRule.RuleType.REJECT, siteLocal.cidr()));
+      assertEquals(204, sync(loopbackClient.server().connectorIpFilterSet("endpoint-default", rules)).getStatus());
```

Comment:
why no returning 200?

Reply:
Because the response is empty "ok" response